catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -929,10 +946,18 @@ private CswRecordCollection queryCsw(GetRecordsType request) throws CswException
             try {
                 queryRequest = queryFactory.updateQueryRequestTags(queryRequest,
                         request.getOutputSchema());
-                LOGGER.debug("Attempting to execute query: {}", queryRequest);
-                QueryResponse queryResponse = framework.query(queryRequest);
+                queryRequest = setQueryPageSize(queryRequest);
```

Comment:
You're correct. This is being discussed out-of-band and will be addressed either in this, or a separate issue.